overlord/snapstate/autorefresh_test.go

```diff
@@ -1165,3 +1165,31 @@ func (s *autoRefreshTestSuite) addRefreshableSnap(names ...string) {
 			}})
 	}
 }
+
+func (s *autoRefreshTestSuite) TestDelayedAutoRefreshIsContinued(c *C) {
+	now := time.Now()
+	restore := snapstate.MockTimeNow(func() time.Time {
+		return now
+	})
+	defer restore()
+
+	af := snapstate.NewAutoRefresh(s.state)
+	s.state.Lock()
+	s.state.Cache("continue-autorefresh", true)
+	s.state.Unlock()
+
+	err := af.Ensure()
+	c.Check(err, IsNil)
+	c.Check(s.store.ops, DeepEquals, []string{"list-refresh"})
+
+	s.state.Lock()
+	defer s.state.Unlock()
+	var lastRefresh time.Time
+	err = s.state.Get("last-refresh", &lastRefresh)
+	c.Assert(err, IsNil)
+	c.Check(lastRefresh.Equal(now), Equals, true)
+
+	var continueAutorefresh bool
+	err = s.state.Get("continue-autorefresh", &continueAutorefresh)
+	c.Check(err, testutil.ErrorIs, &state.NoStateError{})
+}
```

Comment:
Apart from function `canContinueAutorefresh` there is the deferred function that reset cache state `"continue-autorefresh"`. Consider testing `c.Check(s.state.Cached("continue-autorefresh"), Equals, nil)`

Reply:
Thanks, this was a leftover from a previous version. It's fixed now